cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/trigger/AbstractCompositeTrigger.java

```diff
@@ -111,4 +130,15 @@ private void initializeUnitTriggers() {
     }
     return unitTriggerInfos.build();
   }
+
+  /**
+   * Return the simplified version of the current composite trigger based on the condition that the given program
```

Comment:
this description is very hard to understand. Change it to something like:

Create a new trigger where all sub-triggers related to the given program have been removed. Returns null if removing relevant sub-triggers results in a trigger that can never be satisfied.